plumbing/protocol/packp/sideband/demux_test.go

```diff
@@ -0,0 +1,121 @@
+package sideband
+
+import (
+	"bytes"
+	"io/ioutil"
+	"testing"
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/pktline"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type SidebandSuite struct{}
+
+var _ = Suite(&SidebandSuite{})
+
+func (s *SidebandSuite) TestDecode(c *C) {
+	expected := []byte("abcdefghijklmnopqrstuvwxyz")
+
+	buf := bytes.NewBuffer(nil)
+	e := pktline.NewEncoder(buf)
+	e.Encode(append(PackData.Bytes(), expected[0:8]...))
```

Comment:
This is awesome.

I think you should write this line as an example in the comment of the `Channel.Byte()` method, so everybody understand, at a single glance, what is its intended use case.

You can also call the method `func (c Channel) WithPayload(b []byte) []byte`, so its usage will be a little more clear:

```
e.Encode(PackData.WithPayload(expected[0:8])
```

And make it return a multireader to save memory.